apps/users/src/app/app.controller.ts

```diff
@@ -800,6 +824,71 @@ export class UserController {
     }
   }
 
+  /**
+   * User {castcleId} sync social media for update social
+   * @param {CredentialRequest} req Request that has credential from interceptor or passport
+   * @param {SocialSyncDto} body social sync payload
+   * @returns {''}
+   */
+  @UsePipes(new ValidationPipe({ skipMissingProperties: true }))
+  @ApiResponse({
+    status: HttpStatus.NO_CONTENT
+  })
+  @ApiBody({
+    type: SocialSyncDto
+  })
+  @CastleClearCacheAuth(CacheKeyName.SyncSocial)
+  @HttpCode(HttpStatus.NO_CONTENT)
+  @Put('syncSocial')
+  async updateSyncSocial(
+    @Req() req: CredentialRequest,
+    @Body() body: SocialSyncDto
+  ) {
+    logger.log(`Start update sync social.`);
+    logger.log(JSON.stringify(body));
+    const user = await this._getUserFromIdOrCastcleId(body.castcleId, req);
+    if (user) {
+      await this.socialSyncService.update(body, user);
+      return '';
+    } else
+      throw new CastcleException(
+        CastcleStatus.FORBIDDEN_REQUEST,
+        req.$language
+      );
```

Comment:
I think we should return/throw early by writing conditions in form of a guard clause. It's cleaner and easier to read.

```suggestion
    const user = await this._getUserFromIdOrCastcleId(body.castcleId, req);
    
    if (!user) throw new CastcleException(CastcleStatus.FORBIDDEN_REQUEST);
    
    await this.socialSyncService.update(body, user);
```